presto-spi/src/main/java/com/facebook/presto/spi/page/SerializedPage.java

```diff
@@ -104,6 +110,16 @@ public long getChecksum()
         return checksum;
     }
 
+    public String getHostName()
+    {
+        try {
+            return Inet6Address.getLocalHost().getHostName();
```

Comment:
Inet6Address.getLocalHost().getHostAddress() is actually quite expensive. Could potentially be too expensive to be done for every single page. Also I'm not sure if it will return the correct hostname in all cases. What do you think about extracting the current node host name from the node manager (`nodeManager.getCurrentNode().getHostAndPort()`) similarly to how it is done in other places: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaSplitManager.java#L50, https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpLocationFactory.java#L78